src/orders/components/OrderFulfillPage/OrderFulfillPage.tsx

```diff
@@ -340,9 +341,17 @@ const OrderFulfillPage: React.FC<OrderFulfillPageProps> = props => {
                               );
                             }
 
-                            const availableQuantity =
-                              warehouseStock.quantity -
-                              warehouseStock.quantityAllocated;
+                            const allocation: OrderFulfillData_order_allocation = line.allocations.find(
+                              allocation =>
+                                allocation.warehouse.id === warehouse.id
+                            );
+
+                            const availableQuantity = allocation
+                              ? warehouseStock.quantity -
+                                allocation.quantity +
+                                line.quantity
```

Comment:
I'm not sure if it's the right logic here.

Say, warehouse stocks 500 units
Order is placed for 100 units
Saleor allocated 90 units in this warehouse and 10 in other
Saleor allocated 40 units in this warehouse for other orders, so overall allocated quantity for this warehouse equals 130 units

So the quantity available to fulfill equals 500 - 90 +100 = 510, which exceeds warehouse stock quantity.

I believe it should be calculated as 500 - 130 + 90 = 500 - 40 = 460, because it takes into account all stocked units and blocks everything allocated for other orders.

Reply:
Shouldn't any calculation like these be handled in the backend and just returned in API for the client to render?